include/rcutils/types/string_array.h

```diff
@@ -31,6 +31,7 @@ typedef struct RCUTILS_PUBLIC_TYPE rcutils_string_array_t
 {
   size_t size;
   char ** data;
+  rcutils_allocator_t allocator;
```

Comment:
It's not a rule, but it prevents the caller from needing to keep the allocator around and pass it to the `fini` function. This also helps ensure that the same allocator is used with `init` and `fini`.